### DISCUSSION TITLE
Is form validation available yet or planned to be available soon?

### DISCUSSION THREAD

**Discussion by Konstantin-tr** [microsoft/microsoft-ui-xaml#8647] [Q&A]
Asking specifically for usage of interfaces such as INotifyDataErrorInfo, so far I cannot find any resources except for a discussion that it doesn't work yet and a bunch of DIY stuff which requires an insane amount of work to do something as basic as enabling form validation.

In short, is there currently an easy way to get nice red text under my input box when a viewmodel has an error or is this not possible?

**Comment by ChaimCL** [comment]
Through the last community call, it was discovered that they have no plans regarding this for the time being, and there may be something that has a higher priority than these basic controls.

**Reply by Konstantin-tr** [reply]
I think especially because it is basic this should be one of the earliest features available. Form validation is probably the single most common business case for any real world application. Everything needs some level of form validation. That's why every oh-so-small hobby UI framework supports this out of the box. I'm honestly surprised that this does not even have any priority what so ever. Do they know who there audience is for desktop apps? People need validation to work.

**Reply by ChaimCL** [reply]
Yes, in addition to validation, there are inking, maps, data grid and other basic controls that are not implemented, which are in the team's "We are not focused on" section.They seem to be busy cleaning up repository issues and various things with the file explorer team right now

**Comment by lostluna1** [comment]
NEVER
